File: Daily Challenge/C/Reverse String Till Last Vowel 20.4.2026.c
#include <stdio.h>
#include <string.h>

int isVow(char ch){
    return strchr("aeiouAEIOU", ch) != NULL;
}

int main(){
    char str[105];
    scanf("%s", str);

    int n = strlen(str);
    int idx = -1;

    // find last vowel
    for(int i = 0; i < n; i++){
        if(isVow(str[i])){
            idx = i;
        }
    }

    // reverse from 0 to idx
    int i = 0, j = idx;
    while(i < j){
        char temp = str[i];
        str[i] = str[j];
        str[j] = temp;
        i++;
        j--;
    }

    printf("%s", str);
    return 0;
}

File: Daily Challenge/Python/Reverse String Till Last Vowel 20.4.2026.py
def isVow(ch):
    return ch in "aeiouAEIOU"

s = input().strip()
arr = list(s)

idx = -1

# find last vowel
for i in range(len(arr)):
    if isVow(arr[i]):
        idx = i

# reverse from 0 to idx
i, j = 0, idx
while i < j:
    arr[i], arr[j] = arr[j], arr[i]
    i += 1
    j -= 1

print("".join(arr))